src/connection.rs

```diff
@@ -620,7 +620,7 @@ mod tests {
 
         async fn connect(
             &self,
-            _tls_config: Option<Arc<rustls::ClientConfig>>,
+            _tls_config: TlsConfig,
```

Comment:
I was briefly wondering why this code compiled before, but it turns out that this is test code as well and therefore wasn't checked via `cargo build --no-default-features`. Good catch!